core/src/main/scala/chisel3/connectable/Connection.scala

```diff
@@ -234,6 +235,10 @@ private[chisel3] object Connection {
                     deriveChildAlignment(f, proAlign)
                   )
               }
+            // Check that neither consumer nor producer contains probes
+            case (consumer: Data, producer: Data)
+                if (hasProbeTypeModifier(consumer) || hasProbeTypeModifier(producer)) =>
+              errors = "Cannot use connectables with probe types. Omit them prior to connection." +: errors
```

Comment:
Nit: "Exclude" may be more accurate than "omit" given that the APIs to do this are all `exclude`.